scripts/zkai_attention_kv_d32_four_head_seq32_fused_softmax_table_native_gate.py

```
@@ -0,0 +1,1222 @@
+    for part in candidate.parts[1:-1]:
+        current = current / part
+        if current.is_symlink():
+            raise AttentionKvD32FourHeadSeq32FusedSoftmaxTableGateError(
+                f"{label} parent must not be a symlink: {current}"
+            )
+
+
+def reject_symlinked_output_path(path: pathlib.Path, label: str) -> None:
+    candidate = path if path.is_absolute() else pathlib.Path.cwd() / path
+    if candidate.is_symlink():
+        raise AttentionKvD32FourHeadSeq32FusedSoftmaxTableGateError(f"{label} must not be a symlink: {path}")
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)